File: nonlinear equation/dichotomy.py
import numpy as np
def func(x):
    return np.sin(x)
def dic(a,b,e):## 区间（a，b）上的根，e是精度
    while func(a)*func(b)<=0 and abs(a-b)>10e-5:
        c=(a+b)/2
        print('c=',c)
        if func(c)*func(b)<=0:
            a=c
        else:
            b=c
dic(-1,1,10e-5)

File: Nonlinear equation/Dichotomy.java
public class Dichotomy {
    public static void dichotomy(double a,double b,double e){
        double c ;
        while (Function.setF(a)*Function.setF(b)<=0&&Math.abs(a-b)>e){
            c=(a+b)/2;
            System.out.println("c="+c);
            if(Function.setF(c)*Function.setF(b)<=0){
                a=c;
            }else {
                b=c;
            }
    }
}

    public static void main(String[] args) {
        dichotomy(-1,1,10e-5);
    }
}
class Function{
    public static double setF(double x){
        return Math.sin(x);
    }
}
